File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        n = len(s)
        arr = [0]*len(s)
        for i in range(1,n-1):
            if s[i-1]  == s[i] and s[i] == s[i+1]:
                arr[i] = 1
        res = ""
        for i in range(n):
            if arr[i] == 0:
                res += s[i]
        return res


        

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        int count = 1;
        StringBuffer st = new StringBuffer();
        for(int i = 0; i < s.length();i++){
            if(i > 0){
                if(s.charAt(i-1)==s.charAt(i)){
                    count ++;
                }else{
                    count = 1;
                }
            }
            if(count < 3){
                st.append(s.charAt(i));
            }
        }
        return st.toString();
    }
}